packages/webvh/src/cryptosuites/eddsa-jcs-2022.ts

```diff
@@ -0,0 +1,106 @@
+import { type AgentContext, DidDocument } from '@credo-ts/core'
+import {
+  DidsApi,
+  Hasher,
+  Kms,
+  MultiBaseEncoder,
+  TypedArrayEncoder,
+  getPublicJwkFromVerificationMethod,
+} from '@credo-ts/core'
+import { PublicJwk } from '@credo-ts/core/src/modules/kms'
+import { canonicalize } from 'json-canonicalize'
+import { WebVhResource } from '../anoncreds/utils/transform'
+import { ProofOptions } from './types'
+
+export class EddsaJcs2022Cryptosuite {
+  didApi: DidsApi
+  keyApi: Kms.KeyManagementApi
+  agentContext: AgentContext
+  proofOptions: object = {
+    type: 'DataIntegrityProof',
+    cryptosuite: 'eddsa-jcs-2022',
+  }
+  constructor(agentContext: AgentContext) {
+    this.agentContext = agentContext
+    this.didApi = agentContext.dependencyManager.resolve(DidsApi)
+    this.keyApi = agentContext.dependencyManager.resolve(Kms.KeyManagementApi)
+  }
+
+  public async _logError(error: string) {
+    this.agentContext.config.logger.error(error)
+  }
+
+  public async _publicJwkFromId(verificationMethodId: string): Promise<PublicJwk> {
+    const didDocument = new DidDocument(await this.didApi.resolveDidDocument(verificationMethodId))
+    const verificationMethod = didDocument.dereferenceVerificationMethod(verificationMethodId)
+    const publicJwk = getPublicJwkFromVerificationMethod(verificationMethod)
+    return publicJwk
+  }
+
+  public transformation(unsecuredDocument: object, options: ProofOptions) {
+    // https://www.w3.org/TR/vc-di-eddsa/#transformation-eddsa-jcs-2022
+    if (options.type !== 'DataIntegrityProof') {
+      this._logError('PROOF_VERIFICATION_ERROR')
+      throw new Error('PROOF_VERIFICATION_ERROR')
+    }
+    if (options.cryptosuite !== 'eddsa-jcs-2022') {
+      this._logError('PROOF_VERIFICATION_ERROR')
+      throw new Error('PROOF_VERIFICATION_ERROR')
+    }
```

Comment:
Should throw `CredoError` at least, and useful to make the error messsages more descriptive

Reply:
done